turbopack/crates/turbo-tasks-backend/src/backend/operation/aggregation_update.rs

```diff
@@ -2173,6 +2173,7 @@ impl AggregationUpdateQueue {
             TaskDataCategory::Meta,
         );
         let state = get_mut_or_insert_with!(task, Activeness, || ActivenessState::new(task_id));
+        let is_new = state.is_empty();
         let is_positive_now = state.increment_active_counter();
         let is_empty = state.is_empty();
```

Comment:
- I assume there are no issues for `is_new` with negative counts or eventual consistency because the task will eventually be scheduled as active when the count eventually becomes positive.


- `state.is_empty()` may return `false` because `active_counter` is negative. That makes sense for your removal logic, but does it make sense for your `is_new` logic?

https://github.com/vercel/next.js/blob/b81f9892b3f5216e8c40a413f840f6ce8e00f1f4/turbopack/crates/turbo-tasks-backend/src/data.rs#L136

should the check be `<= 0` instead of `== 0`?

Reply:
I think there is indeed something wrong here.

The existence of `Activeness` makes a task active, even when the active_counter is negative.

So we also need to schedule the task in `decrease_active_count` when the `Activeness` didn't exist before. I'll add that.